### PR TITLE
Add support for nested interfaces in @ConfigProperties

### LINKED ISSUE #10511: Support configuration with nested interfaces
**Description**
Actually it's possible to have nested configuration classes. It would be nice to have this opportunity for interfaces too.

Something like this:

```
@ConfigProperties(prefix = "acme")
public interface MyConfig {

    Greeting getGreeting();

    @ConfigProperties
    interface Greeting {
        String getHelloString();
    }
}
```

### PR DESCRIPTION
**Pull request by geoand**
Resolves: #10511

### CODE REVIEW

**Comment by geoand** [comment]
The Gradle CI failure is completely unrelated

**Comment by geoand** [comment]
@mkouba if you have a few minutes to review this I would be very grateful since if changes are needed, I won't be able to make them for a few days (I'll be on PTO)

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesBuildStep.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package io.quarkus.arc.deployment.configproperties;
 
+import static io.quarkus.arc.deployment.configproperties.InterfaceConfigPropertiesUtil.*;
```
Comment (mkouba) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesBuildStep.java:3]: I thought that we should not use wildcard in imports, or?

**Review by mkouba — APPROVED** [review]
Looks good. Except that it's not about _sub-interfaces_ but _nested_ interfaces ;-).

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesBuildStep.java (hunk 1 above)
Comment (geoand) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesBuildStep.java:3]: Ah darn... I'll fix it

**Comment by geoand** [comment]
Right, I'm always terrible at naming, let me fix the names :)

**Pull request merged**